src/ledger/database/lmdb.zig

```diff
@@ -0,0 +1,641 @@
+const std = @import("std");
+const c = @import("lmdb");
+const sig = @import("../../sig.zig");
+const database = @import("lib.zig");
+const build_options = @import("build-options");
+
+const Allocator = std.mem.Allocator;
+
+const BytesRef = database.interface.BytesRef;
+const ColumnFamily = database.interface.ColumnFamily;
+const IteratorDirection = database.interface.IteratorDirection;
+const Logger = sig.trace.Logger;
+
+const key_serializer = database.interface.key_serializer;
+const value_serializer = database.interface.value_serializer;
+
+pub fn LMDB(comptime column_families: []const ColumnFamily) type {
```

Comment:
The RocksDB implementation has the logger as part of the struct's states. Any reason not to have same for the `LMDB`? 

Also, the logger can be scoped.

Reply:
The only reason rocksdb gets a logger is to communicate its errors through strings. Any time it returns an error, it also logs an error message, so the cause of the error is clear. LMDB only communicates errors through various error codes, which can be represented with zig errors.